tests/collector/utils/requestUtils.js

```diff
@@ -0,0 +1,272 @@
+/**
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/collector/interceptRequests.js
+ */
+'use strict'; // eslint-disable-line strict
+const expect = require('chai').expect;
+const sinon = require('sinon');
+const interceptor = require('express-interceptor');
+const Promise = require('bluebird');
+const supertest = require('supertest');
+const app = require('../../../index').app;
+const tu = require('../../testUtils');
+const forkUtils = require('./forkUtils');
+const constants = require('../../../api/v1/constants');
+const genPath = '/v1/generators';
+const api = supertest(app);
+supertest.Test.prototype.end = Promise.promisify(supertest.Test.prototype.end);
+supertest.Test.prototype.then = function (resolve, reject) {
+  return this.end().then(resolve).catch(reject);
+};
+
+let token;
+
+module.exports = {
+  setupInterception,
+  doStart,
+  postGenerator,
+  patchGenerator,
+  putGenerator,
+  postStatus,
+  getStatus,
+  getCollector,
+  setToken,
+  expectHeartbeatGens,
+  expectHeartbeatStatus,
+  expectSubjectQuery,
+  expectBulkUpsertNames,
+  expectBulkUpsertSamples,
+};
+
+function setupInterception(interceptConfig) {
+  setupMiddleware(interceptConfig);
+  setupInterceptFuncs(interceptConfig);
+}
+
+function setupMiddleware(interceptConfig) {
+  app.use(interceptor(function (req, res) {
+    const regexes = Object.values(interceptConfig).map(v => v.path);
+    return {
+      isInterceptable: () => regexes.some((re) => req.path.match(re)),
+      intercept: (body, send) => {
+        res.body = JSON.parse(body);
+        send(body);
+      },
+    };
+  }));
+}
+
+function setupInterceptFuncs(interceptConfig) {
+  Object.entries(interceptConfig).forEach(([reqType, conf]) => {
+    conf.reqType = reqType;
+    conf.promiseMap = { '': [] };
+    conf.timeoutMap = { '': [] };
+
+    interceptRequest(conf);
+    const awaitFunc = awaitRequest.bind(null, conf);
+    if (conf.expectedInterval) {
+      const tickAndAwait = forkUtils.tickUntilComplete.bind(null, awaitFunc);
+      module.exports[`await${reqType}`] = tickAndAwait;
+    } else {
+      module.exports[`await${reqType}`] = awaitFunc;
+    }
+  });
+
+  function interceptRequest(conf) {
+    const { controller, method } = conf;
+    if (controller[method].restore) controller[method].restore();
+    const originalMethod = controller[method];
+
+    sinon.stub(controller, method)
+    .callsFake(callThenResolve.bind(null, originalMethod, conf));
+  }
+
+  function awaitRequest(conf, collectorName='') {
+    const timeout = conf.expectedInterval && conf.expectedInterval() * 1.5;
```

Comment:
what is the default value if conf.expectedInterval is undefined?

Reply:
If it's undefined no timeout will be set.